### ISSUE TITLE
ConcurrentModificationException during Command use save

### ISSUE THREAD

**Issue by Rsl1122** [Rsl1122/Plan-PlayerAnalytics#110]
**Plan Version:** 3.3.0
**Server Version:** paper-1.11.2
**Database Type:** Unknown
**Command Causing Issue:** None

**Description:**
ConcurrentModificationException during Command use save
(Command was issued during save)

**Steps to Reproduce:**

1. Issue command while commanduse is being saved

**Proposed Solution:**
One line fix:
Copy map on save

**Stack Trace - Console/Errors.txt contents:**
```
[12:16:05] [Craft Scheduler Thread - 2372/WARN]: [Plan] Plugin Plan v3.3.0 generated an exception while executing task 1226
java.util.ConcurrentModificationException
    at java.util.HashMap$HashIterator.nextNode(HashMap.java:1437) ~[?:1.8.0_131]
    at java.util.HashMap$KeyIterator.next(HashMap.java:1461) ~[?:1.8.0_131]
    at main.java.com.djrapitops.plan.database.tables.CommandUseTable.saveCommandUse(CommandUseTable.java:95) ~[?:?]
    at main.java.com.djrapitops.plan.database.Database.saveCommandUse(Database.java:199) ~[?:?]
    at main.java.com.djrapitops.plan.data.cache.DataCacheHandler.saveCommandUse(DataCacheHandler.java:341) ~[?:?]
    at main.java.com.djrapitops.plan.data.cache.DataCacheHandler$1.run(DataCacheHandler.java:151) ~[?:?]
    at org.bukkit.craftbukkit.v1_11_R1.scheduler.CraftTask.run(CraftTask.java:58) ~[patched_1.11.2.jar:git-Paper-1104]
    at org.bukkit.craftbukkit.v1_11_R1.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:52) [patched_1.11.2.jar:git-Paper-1104]
    at com.destroystokyo.paper.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:22) [patched_1.11.2.jar:git-Paper-1104]
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_131]
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_131]
    at java.lang.Thread.run(Thread.java:748) [?:1.8.0_131]
```

**Comment by FrozenBeard** [comment]
Thanks for making a ticket for me! Database type is MariaDB, Its local too so im hitting 127.0.0.1. Shouldn't be any latency, unless I screwed up my settings somehow or I have some large i/o. Wouldn't surprise me since im using a HDD.

So this is something that's fixable from your end?

Ill keep the plug-in updated and running as its collecting data!

**Comment by Rsl1122** [comment]
This Exception has been fixed in version 3.4.1 :)

> Fixed ConcurrentModificationException if a command was used while commanduse was being saved.

https://www.spigotmc.org/resources/plan-player-analytics.32536/update?update=165809

----

If you're talking about the command timing out, next version will bring a better debug log to file option so will have to wait until that is live to look into the timeout issue.
